### PR TITLE
DNM: os/bluestore: add libzbc as submodule. added flag WITH_LIBZBC

### PR DESCRIPTION
**Pull request by shehbazj**
added flag WITH_LIBZBC in cmake command to build libzbc
added libzbc module

Signed-off-by: Shehbaz Jaffer shehbaz.jaffer@mail.utoronto.ca

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -547,10 +547,21 @@ endif(HAVE_ARMV8_CRC)
 
 add_library(common_utf8 STATIC common/utf8.c)
 
+if(${WITH_LIBZBC})
```
Comment (tchaikov) [inline src/CMakeLists.txt]: might want to add an option `WITH_LIBZBC` for this.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -547,10 +547,21 @@ endif(HAVE_ARMV8_CRC)
...
+	message("Adding LIBZBC support")
+  add_subdirectory(libzbc)
+endif(${WITH_LIBZBC})
+
 if(${WITH_LTTNG})
   add_subdirectory(tracing)
 endif(${WITH_LTTNG})
 
+if(${WITH_LIBZBC})
+  set(LIBZBC_INCLUDE_PATH,libzbc/include)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: syntax error. see https://cmake.org/cmake/help/v3.0/command/set.html .

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -547,10 +547,21 @@ endif(HAVE_ARMV8_CRC)
...
+  include_directories(${LIBZBC_INCLUDE_PATH})
```
Comment (tchaikov) [inline src/CMakeLists.txt]: better off using `target_include_directories()` if libzbc is only used by a target.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -547,10 +547,21 @@ endif(HAVE_ARMV8_CRC)
...
+  target_link_libraries(zbc)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: syntax error. see https://cmake.org/cmake/help/v3.0/command/target_link_libraries.html

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -811,8 +826,11 @@ if(WITH_LIBCEPHFS)
     install(TARGETS ceph-fuse DESTINATION bin)
     install(PROGRAMS mount.fuse.ceph DESTINATION ${CMAKE_INSTALL_SBINDIR})
   endif(HAVE_LIBFUSE)
+
 endif(WITH_LIBCEPHFS)
 
+
```
Comment (tchaikov) [inline src/CMakeLists.txt]: might want to remove these empty lines.

File: src/CMakeLists.txt [hunk 6]
```diff
@@ -798,6 +809,10 @@ if(WITH_LIBCEPHFS)
     $<TARGET_OBJECTS:common_mountcephfs_objs>)
   target_link_libraries(mount.ceph keyutils)
 
+	if(${WITH_LIBZBC})
```
Comment (tchaikov) [inline src/CMakeLists.txt]: could you use spaces instead of tab in the cmake script?

File: src/CMakeLists.txt [hunk 7]
```diff
@@ -798,6 +809,10 @@ if(WITH_LIBCEPHFS)
...
+		target_link_libraries(cephfs zbc)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: are we using libzbc in cephfs now? if not, could you postpone this commit until we are?

File: .gitmodules [hunk 8]
```diff
@@ -33,3 +33,6 @@
 [submodule "src/isa-l"]
 	path = src/isa-l
 	url = https://github.com/ceph/isa-l
+[submodule "src/libzbc"]
```
Comment (amitkumar50) [inline .gitmodules]: @shehbazj Please keep PR title < 50 characters
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

**Pull request closed without merging**